skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
```

Comment:
OK, sorry I could not work on this anymore the last weeks. 

So, I chose option 1. We cannot really imitate Matlab for this anymore, because the parameter `levels` does unfortunately not match neither of the two parameters `GrayLimits` and `NumLevels`. If I wanted to match the behavior of `NumLevels`, the function would really change w.r.t what it does now. 

I make a new pull request ...